pkg/plugin/processor/builtin/unwrap_debezium.go

```diff
@@ -0,0 +1,347 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=unwrap_debezium_paramgen.go unwrapDebeziumConfig
+
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/foundation/multierror"
+	"github.com/goccy/go-json"
+)
+
+const (
+	debeziumOpCreate = "c"
+	debeziumOpUpdate = "u"
+	debeziumOpDelete = "d"
+	debeziumOpRead   = "r"      // snapshot
+	debeziumOpUnset  = "$unset" // mongoDB unset operation
+
+	debeziumFieldBefore    = "before"
+	debeziumFieldAfter     = "after"
+	debeziumFieldSource    = "source"
+	debeziumFieldOp        = "op"
+	debeziumFieldTimestamp = "ts_ms"
+)
+
+type unwrapDebeziumConfig struct {
+	// Field is a reference to the field which contains the Debezium record.
+	//
+	// For more information about record references, see: https://github.com/ConduitIO/conduit-processor-sdk/blob/cbdc5dcb5d3109f8f13b88624c9e360076b0bcdb/util.go#L66.
+	Field string `json:"field" validate:"regex=^.Payload" default:".Payload.After"`
+}
+
+type unwrapDebezium struct {
+	sdk.UnimplementedProcessor
+
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func newUnwrapDebezium(logger log.CtxLogger) sdk.Processor {
+	return &unwrapDebezium{logger: logger}
+}
+
+func (u *unwrapDebezium) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.debezium",
+		Summary: "Unwraps a Debezium record from the input OpenCDC record.",
+		Description: `This processor unwraps a Debezium record from the input OpenCDC record.
+
+This is useful in cases where Conduit acts as an intermediary between a Debezium source and a Debezium destination. 
+In such cases, the Debezium record is set as the OpenCDC record's payload, and needs to be unwrapped for further usage.`,
```

Comment:
We should probably mention that the resulted unwrapped record will completely replace the current record. Only the position will be copied over to the new record. We should also explain how metadata is handled (the debezium metadata and record metadata is merged, on conflicts debezium metadata takes precedence).